packages/package_info/lib/package_info.dart

```diff
@@ -4,19 +4,59 @@
 
 import 'dart:async';
 
+import 'package:meta/meta.dart';
 import 'package:flutter/services.dart';
 
-/// Provides information about the current application package.
 const MethodChannel _kChannel =
     const MethodChannel('plugins.flutter.io/package_info');
 
-/// Returns the `CFBundleShortVersionString` on iOS or `versionName` on Android
-Future<String> get version async => await _kChannel.invokeMethod('getVersion');
+/// Wrap application Bundle (on iOS) and Package (on Android) information,
```

Comment:
Suggestions:

- Let the class-level DartDoc start out by saying what concept the class represents. Application package information, in this case. Then what that amounts to in platform-specific terms on iOS and Android.
- I find the mention of caching in lines 14,16-17 confusing. AFAIK, we are talking about static information retrieved from the application bundle. If the information could change at runtime, we should reconsider the API.
- Don't mention each piece of information here. We'll most likely forget to update it, if we add another piece. The needed documentation is already present, in more detail, on each property below.
- Avoid discussing the implementation. The fact that `PackageInfo` is a singleton is either obvious from the `getInstance()` API, or unimportant/slightly confusing as the `getInstance()` method is async.

Reply:
dartdoc(from dart 1.24.3) always failed with:
```
Generation failed: Unknown type GenericFunctionTypeElementImpl
package:dartdoc/src/model.dart 1528                                  new ModelElement.from
package:dartdoc/src/model.dart 2607                                  new TopLevelVariable
package:dartdoc/src/model.dart 1306                                  Library._getVariables.<fn>
dart:core                                                            Iterable.toList
package:dartdoc/src/model.dart 1307                                  Library._getVariables
package:dartdoc/src/model.dart 1099                                  Library.constants
package:dartdoc/src/model.dart 1344                                  Library.allModelElements
package:dartdoc/src/model.dart 2396                                  Package.allModelElements.<fn>
dart:core                                                            List.forEach
package:dartdoc/src/model.dart 2395                                  Package.allModelElements
package:dartdoc/dartdoc.dart 171                                     DartDoc.generateDocs
/Users/whesse/dart/sdk/third_party/pkg/dartdoc/bin/dartdoc.dart 156  main.<fn>
package:stack_trace                                                  Chain.capture
/Users/whesse/dart/sdk/third_party/pkg/dartdoc/bin/dartdoc.dart 155  main
===== asynchronous gap ===========================
dart:async                                                           _Completer.completeError
package:dartdoc/dartdoc.dart 191                                     DartDoc.generateDocs
===== asynchronous gap ===========================
dart:async                                                           new Future.microtask
package:dartdoc/dartdoc.dart 130                                     DartDoc.generateDocs
/Users/whesse/dart/sdk/third_party/pkg/dartdoc/bin/dartdoc.dart 156  main.<fn>
===== asynchronous gap ===========================
dart:async                                                           new Future.microtask
/Users/whesse/dart/sdk/third_party/pkg/dartdoc/bin/dartdoc.dart 155  main.<fn>
package:stack_trace                                                  Chain.capture
/Users/whesse/dart/sdk/third_party/pkg/dartdoc/bin/dartdoc.dart 155  main
```

Can I fix the doc in next pull request?